File: algorithms/232. Implement Queue using Stacks/solution.py
class Queue(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self._data = []
        

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        self._data.append(x)

    def pop(self):
        """
        :rtype: nothing
        """
        self._data.pop(0)

    def peek(self):
        """
        :rtype: int
        """
        return self._data[0]

    def empty(self):
        """
        :rtype: bool
        """
        return len(self._data) == 0

File: algorithms/232. Implement Queue using Stacks/solution.cc
class MyQueue {
    stack<int> in;
    stack<int> out;
public:
    /** Initialize your data structure here. */
    MyQueue() {
        
    }
    
    /** Push element x to the back of queue. */
    void push(int x) {
        in.push(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        int val = peek();
        out.pop();
        return val;
    }
    
    /** Get the front element. */
    int peek() {
        if ( out.empty() ) {
            while ( !in.empty() ) {
                out.push(in.top());
                in.pop();
            }
        }
        return out.top();
    }
    
    /** Returns whether the queue is empty. */
    bool empty() {
        return in.empty() && out.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * bool param_4 = obj.empty();
 */
